Integrations/src/main/java/io/deephaven/integrations/numpy/Java2NumpyCopy.java

```diff
@@ -248,6 +265,149 @@ public static void copySlice(final Table t, final long rowStart, final long[] da
+    }
+
+    /**
+     * Casts data to the desired type and copies a slice of rows into a flattened 2D array.
+     * This is useful for copying table data directly into numpy arrays.
+     *
+     * @param t        table to copy data from
+     * @param rowStart first row of data to copy
+     * @param data     array to copy data into
+     * @param nRow     number of rows to copy; also the number of rows in <code>data</code>.
+     * @param nCol     number of table columns; also the number of columns in <code>data</code>.
+
+    public static void copySubset(final Table t, final long rowStart, final short[] data, final int nRow, final int nCol) {
+        assertCopySliceArgs(t, rowStart, data == null ? -1 : data.length, nRow, nCol);
+        assert data != null;
+        copySubset(t, rowStart, nRow, nCol, short.class, "(short)", (cs, k, idx) -> data[idx] = cs.getShort(k));
+    }
+
+    /**
+     * Casts data to the desired type and copies a slice of rows into a flattened 2D array.
+     * This is useful for copying table data directly into numpy arrays.
+     *
+     * @param t        table to copy data from
+     * @param rowStart first row of data to copy
+     * @param data     array to copy data into
+     * @param nRow     number of rows to copy; also the number of rows in <code>data</code>.
+     * @param nCol     number of table columns; also the number of columns in <code>data</code>.
+
+    public static void copySubset(final Table t, final long rowStart, final int[] data, final int nRow, final int nCol) {
+        assertCopySliceArgs(t, rowStart, data == null ? -1 : data.length, nRow, nCol);
+        assert data != null;
+        copySubset(t, rowStart, nRow, nCol, int.class, "(int)", (cs, k, idx) -> data[idx] = cs.getInt(k));
+    }
+
+    /**
+     * Casts data to the desired type and copies a slice of rows into a flattened 2D array.
+     * This is useful for copying table data directly into numpy arrays.
+     *
+     * @param t        table to copy data from
+     * @param rowStart first row of data to copy
+     * @param data     array to copy data into
+     * @param nRow     number of rows to copy; also the number of rows in <code>data</code>.
+     * @param nCol     number of table columns; also the number of columns in <code>data</code>.
+
+    public static void copySubset(final Table t, final long rowStart, final long[] data, final int nRow, final int nCol) {
+        assertCopySliceArgs(t, rowStart, data == null ? -1 : data.length, nRow, nCol);
+        assert data != null;
+        copySubset(t, rowStart, nRow, nCol, long.class, "(long)", (cs, k, idx) -> data[idx] = cs.getLong(k));
+    }
+*/
+// #####################################################################################################################
+// # End the new stuff
+// #####################################################################################################################
```

Comment:
1) The start and end comments are not appropriate.  All new code should look like it is part of the whole and not some addition.

2) I don't think these new methods are doing what you want.  I assume you want to do something quite different.  Maybe something like:
` public static void fillNumpy(final ColumnSource<>[] cols, final IndexSet indices, final long[] data, final int nRow, final int nCol)`